### PR TITLE
handler returns error message as json

### PR DESCRIPTION
**Pull request by tinymarsracing**

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -44,15 +44,17 @@ const handler = (req, res) => {
     if (err) {
       // eslint-disable-next-line no-console
       console.error(err);
-      res.writeHead(err.status || err.statusCode || 500);
-      res.end(err.message);
+      res.statusCode = err.status || err.statusCode || 500;
+      res.setHeader('Content-Type', 'application/json');
+      res.end(JSON.stringify({ error: err.message }));
```
Comment (benjie) [inline src/index.js]: For GraphQL compatibility, this should be
```suggestion
      res.end(JSON.stringify({ errors: [{message: err.message}] }));
```

File: src/index.js [hunk 2]
```diff
@@ -44,15 +44,17 @@ const handler = (req, res) => {
...
       return;
     }
     if (!res.finished) {
       if (!res.headersSent) {
-        res.writeHead(404);
+        res.statusCode = 404;
       }
-      res.end(`'${req.url}' not found`);
+      res.setHeader('Content-Type', 'application/json');
+      res.end(JSON.stringify({ error: `'${req.url}' not found` }));
```
Comment (benjie) [inline src/index.js]: This one probably doesn't need GraphQL compatibility because it's not the GraphQL endpoint.

File: src/index.js [hunk 3]
```diff
@@ -44,15 +44,17 @@ const handler = (req, res) => {
...
+      res.end(JSON.stringify({ error: err.message }));
       return;
     }
     if (!res.finished) {
       if (!res.headersSent) {
-        res.writeHead(404);
+        res.statusCode = 404;
       }
-      res.end(`'${req.url}' not found`);
+      res.setHeader('Content-Type', 'application/json');
```
Comment (benjie) [inline src/index.js]: This needs to be in the `if` block; if `res.headersSent` is true then we cannot safely do `res.setHeader`.

**Review by benjie — CHANGES_REQUESTED** [review]
Thanks for this; a couple of tweaks and it's good to merge I think! 👍

**Comment by tinymarsracing** [comment]
Okay, thanks for reviewing, I implemented the changes.

**Review by benjie — APPROVED** [review]
Thanks!

**Pull request merged**